src/libs/SidebarUtils.js

```diff
@@ -86,85 +86,90 @@ Onyx.connect({
+    const reportsToDisplay = _.filter(reports, report => ReportUtils.shouldReportBeInOptionList(report, reportIDFromRoute, isInGSDMode, currentUserLogin, iouReports, betas, policies));
 
     // Get all the display names for our reports in an easy to access property so we don't have to keep
     // re-running the logic
-    const filteredReportsWithReportName = _.map(filteredReports, (report) => {
+    _.each(reportsToDisplay, (report) => {
         // Normally, the spread operator would be used here to clone the report and prevent the need to reassign the params.
         // However, this code needs to be very performant to handle thousands of reports, so in the interest of speed, we're just going to disable this lint rule and add
         // the reportDisplayName property to the report object directly.
         // eslint-disable-next-line no-param-reassign
         report.reportDisplayName = ReportUtils.getReportName(report, policies);
-        return report;
     });
 
-    // Sorting the reports works like this:
-    // - When in default mode, reports will be ordered by most recently updated (in descending order) so that the most recently updated are at the top
-    // - When in GSD mode, reports are ordered by their display name so they are alphabetical (in ascending order)
-    // - Regardless of mode, all archived reports should remain at the bottom
-    const orderedReports = _.sortBy(filteredReportsWithReportName, (report) => {
-        if (ReportUtils.isArchivedRoom(report)) {
-            return isInDefaultMode
-
-                // -Infinity is used here because there is no chance that a report will ever have an older timestamp than -Infinity and it ensures that archived reports
-                // will always be listed last
-                ? -Infinity
-
-                // Similar logic is used for 'ZZZZZZZZZZZZZ' to reasonably assume that no report will ever have a report name that will be listed alphabetically after this, ensuring that
-                // archived reports will be listed last
-                : 'ZZZZZZZZZZZZZ';
+    // The LHN is split into five distinct groups, and each group is sorted a little differently. The groups will ALWAYS be in this order:
+    // 1. Pinned - Always sorted by reportDisplayName
+    // 2. Outstanding IOUs - Always sorted by iouReportAmount with the largest amounts at the top of the group
+    // 3. Drafts - Always sorted by reportDisplayName
+    // 4. Non-archived reports
+    //      - Sorted by lastMessageTimestamp in default (most recent) view mode
+    //      - Sorted by reportDisplayName in GSD (focus) view mode
+    // 5. Archived reports
+    //      - Sorted by lastMessageTimestamp in default (most recent) view mode
+    //      - Sorted by reportDisplayName in GSD (focus) view mode
+    const reportGroups = _.groupBy(reportsToDisplay, (report) => {
```

Comment:
Grouping these into an object keyed with these strings is less clear than something like setting up a bunch of empty arrays:

```js
const pinnedReports = [];
const outstandingIOUReports = [];
```

I forgot what `groupBy` does, wasn't really sure what was happening here and had to look it up.

Could we just set up the arrays, then sort them, then concat them?

Reply:
Yes, that is a good idea and hopefully it will be more clear that way. I've update the PR for that.